clang/test/CodeGenSYCL/int_header1.cpp

```diff
@@ -4,14 +4,14 @@
 // CHECK:template <> struct KernelInfo<class KernelName> {
 // CHECK:template <> struct KernelInfo<::nm1::nm2::KernelName0> {
 // CHECK:template <> struct KernelInfo<::nm1::KernelName1> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3< ::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3< ::nm1::KernelName1>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4< ::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4< ::nm1::KernelName1>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3<KernelName5>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4<KernelName7>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName8< ::nm1::nm2::C>> {
-// CHECK:template <> struct KernelInfo<class TmplClassInAnonNS<class ClassInAnonNS>> {
+// CHECK:template <> struct KernelInfo<nm1::KernelName3<nm1::nm2::KernelName0>> {
```

Comment:
Why we can stop emit global namespace here?

Reply:
It was emitted earlier because NameType was emitted as Ty.getAsString() earlier (and this kept all the global namespace specifiers). 

When the NameType is a template specialization type it is now necessary to emit the template name and then loop through arguments individually, printing them individually. I used printFullyQualifiedName to emit template name and  Arg.print to print the template arguments. Both don't emit the global namespace specifier. I was not sure if the global access specifier was actually required or whether it just happened to be emitted because Ty.getAsString emits it. So I ran tests to see if anything failed without it and nothing did. Is it actually required?

Reply:
>>So I ran tests to see if anything failed without it and nothing did. Is it actually required?

By tests, I mean check-sycl tests and other regression tests.

Reply:
Looking into this commit message https://github.com/intel/llvm/commit/5d22a8ee96e9addcfb28c79b4b1c84e79ff3ca04, I guess it was emitted on purpose. We just missed tests which really check described case. But this is the case when user adds something into sycl namespace, I guess if the spec will explicitly prohibit it we can elide this global namespace specifier here.
But while we don't have fixed spec, I wouldn't remove global namespace specifier because it could be someone's workflow.

Reply:
Ok Thank you. I'll try figuring out how to force it.

Reply:
Global namespace has scope resolution operator emitted now. The lit tests still need to be modified because of extra space. I didn't fix the spacing because it wasn't consistent in integration header anyway.